File: LeetCode/48.Rotate Image/python/main.py
class Solution:
    def transpose(self, arr):
        for i in range(len(arr)):
            for j in range(i):
                temp = arr[i][j]
                arr[i][j] = arr[j][i]
                arr[j][i] = temp

    def reverse(self, arr):
        for r in range(len(arr)):
            left = 0
            right = len(arr) - 1
            while left < right:
                temp = arr[r][left]
                arr[r][left] = arr[r][right]
                arr[r][right] = temp
                left += 1
                right -= 1

    def rotate(self, matrix):
        self.transpose(matrix)
        self.reverse(matrix)

def main():
    solution = Solution()
    arr = [[1,2,3],[4,5,6],[7,8,9]] #Example case hardcoded
    solution.rotate(arr)
    print(arr)

if __name__ =="__main__":
    main()

File: LeetCode/48.Rotate Image/c++/main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();
        
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                swap(matrix[i][j], matrix[j][i]);
            }
        }
        
        for (int i = 0; i < n; ++i) {
            reverse(matrix[i].begin(), matrix[i].end());
        }
    }
};

int main(){
    Solution solution;
    vector<vector<int>> arr = {{1,2,3},{4,5,6},{7,8,9}};
    cout<<"Initial array is:-"<<endl;
    for (int i = 0; i < arr.size(); i++)
    {
        for (int j = 0; j < arr[i].size(); j++)
        {
            cout<<arr[i][j]<<" ";
        }
        cout<<endl;
    }
    solution.rotate(arr);
    
    cout<<"Final array is:-"<<endl;
    for (int i = 0; i < arr.size(); i++)
    {
        for (int j = 0; j < arr[i].size(); j++)
        {
            cout<<arr[i][j]<<" ";
        }
        cout<<endl;
    }
    
}